pkg/resiliency/endpoint/bpf_backer.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package endpoint
+
+import (
+	"fmt"
+
+	"github.com/vishvananda/netlink"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+)
+
+// EndpointBPFProgBacker wraps a given endpoint.
+type EndpointBPFProgBacker struct {
+	ep *endpoint.Endpoint
+}
+
+// NewEndpointBPFProgBacker returns a new instance.
+func NewEndpointBPFProgBacker(ep *endpoint.Endpoint) *EndpointBPFProgBacker {
+	return &EndpointBPFProgBacker{ep: ep}
+}
+
+// GetState fetches the current endpoint state.
+func (e *EndpointBPFProgBacker) GetState() endpoint.State {
+	return e.ep.GetState()
+}
+
+// String returns a human readable string.
+func (e *EndpointBPFProgBacker) HostInterface() string {
+	return e.ep.HostInterface()
+}
+
+// String returns a human readable string.
+func (e *EndpointBPFProgBacker) String() string {
+	return fmt.Sprintf("[%s] %s(%d)", e.ep.HostInterface(), e.ep.K8sPodName, e.ep.ID)
+}
+
+// GetLink retrieves the associated link.
+func (e *EndpointBPFProgBacker) GetLink() (netlink.Link, error) {
+	return netlink.LinkByName(e.ep.HostInterface())
+}
+
+// QdiscList retrieves associated qdisc attrs.
+func (e *EndpointBPFProgBacker) QdiscList(l netlink.Link) ([]netlink.Qdisc, error) {
+	return netlink.QdiscList(l)
```

Comment:
These are linux-specific and thus should live somewhere under `pkg/datapath/linux`. This should also be marked with `linux` build tag.

Reply:
Thanks for the feedback Jussi! I've reworked the pr to line up with your expectations.